src/main/java/com/commercetools/sync/products/utils/ProductVariantPriceUpdateActionUtils.java

```diff
@@ -0,0 +1,114 @@
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Optional;
+
+import static com.commercetools.sync.commons.utils.CommonTypeUpdateActionUtils.buildUpdateAction;
+
+public final class ProductVariantPriceUpdateActionUtils {
+
+    /**
+     * Compares all the fields of a {@link Price} and a {@link PriceDraft} and returns a list of
+     * {@link UpdateAction}&lt;{@link Product}&gt; as a result. If both the {@link Price} and the {@link PriceDraft}
+     * have identical fields, then no update action is needed and hence an empty {@link List} is returned.
+     *
+     * @param variantId   the variantId needed for building the update action.
+     * @param oldPrice    the price which should be updated.
+     * @param newPrice    the price draft where we get the new fields.
+     * @param syncOptions responsible for supplying the sync options to the sync utility method. It is used for
+     *                    triggering the error callback within the utility, in case of errors.
+     * @return A list with the update actions or an empty list if the price fields are identical.
+     */
+    @Nonnull
+    public static List<UpdateAction<Product>> buildActions(
+        @Nonnull final Integer variantId,
+        @Nonnull final Price oldPrice,
+        @Nonnull final PriceDraft newPrice,
+        @Nonnull final ProductSyncOptions syncOptions) {
+
+        final List<UpdateAction<Product>> updateActions = new ArrayList<>();
+
+        buildChangePriceUpdateAction(oldPrice, newPrice).ifPresent(updateActions::add);
+        updateActions.addAll(buildCustomUpdateActions(variantId, oldPrice, newPrice, syncOptions));
+
+        return updateActions;
+    }
+
+
+    /**
+     * Builds a {@link ChangePrice} action based on the comparison of the following fields of the supplied {@link Price}
+     * and {@link PriceDraft}:
+     * <ul>
+     *     <li>{@link Price#getValue()} and {@link PriceDraft#getValue()}</li>
+     *     <li>{@link Price#getTiers()} and {@link PriceDraft#getTiers()}</li>
+     * </ul>
+     *
+     * <p>If any of the aforementioned fields are different a {@link ChangePrice} update action will be returned in an
+     * {@link Optional}, otherwise if both are identical in the {@link Price} and the {@link PriceDraft}, then no update
+     * action is needed and hence an empty {@link Optional} is returned.
+     *
+     * @param oldPrice  the price which should be updated.
+     * @param newPrice  the price draft where we get the new name.
+     * @return A filled optional with the update action or an empty optional if the names are identical.
+     */
+    @Nonnull
+    public static Optional<ChangePrice> buildChangePriceUpdateAction(
+        @Nonnull final Price oldPrice,
+        @Nonnull final PriceDraft newPrice) {
+
+        final Optional<ChangePrice> actionAfterValuesDiff = buildUpdateAction(oldPrice.getValue(), newPrice.getValue(),
+            () -> ChangePrice.of(oldPrice, newPrice, true));
```

Comment:
since `value` is a required field within Price (https://docs.commercetools.com/http-api-projects-products.html#pricedraft), shouldn't we validate that the `newPrice.getValue()` is not null? if so, should't we call to the warning callback of `ProductSyncOptions`?

Reply:
good catch! I missed that. 2b80d56

Thanks :)